### ISSUE TITLE
ILD reco tutorial differences between zsh and bash

### ISSUE THREAD

**Issue by alaudrain** [key4hep/key4hep-tutorials#16]
Hello!

I'm trying to run the ILD reconstruction tutorial but I'm running into an error at the last step: simulation (using `ddsim`) works fine, but at the reconstruction step (using `k4run ILDReconstruction.py`) I see the multiple instances of the following messages.

```
[ ERROR "MyAdd4MomCovMatrixCharged"]  Warning: null charge particle! This case is not supported
```
and towards the end of the execution (after the event loop):
```
Failed to create calo hit: STATUS_CODE_NOT_FOUND
PandoraApi::CaloHit::Create(m_pandora, caloHitParameters) throw STATUS_CODE_NOT_FOUND
    in function: CreateMuonCaloHits
    in file:     /tmp/root/spack-stage/spack-stage-ddmarlinpandora-0.12.01-p742e5lzpihslq2zfjsto66iki2aaskn/spack-src/src/DDCaloHitCreator.cc line#: 388
[ ERROR "MyDDMarlinPandora"] Failed to extract muon hit: STATUS_CODE_NOT_FOUND
EDM4hepOutput       ERROR Could not write event, because the following collections are not present: V0RecoParticles, V0RecoParticles_particleIDs, V0Vertices
EDM4hepOutput       ERROR Maximum number of errors ( 'ErrorMax':1) reached.
EventLoopMgr      WARNING Execution of algorithm EDM4hepOutput failed
EventLoopMgr        ERROR Error processing event loop.
EventLoopMgr        ERROR Terminating event processing loop due to errors
EventLoopMgr        ERROR Terminating event processing loop due to errors
```
and finally the last line of the output
```
ApplicationMgr      ERROR Application Manager Terminated with error code 3
```

I noticed that the files are still created though. Now the actually surprising thing is, if I run the EXACT same setup using bash instead of zsh (I'm using zsh by default), I only see the following message:
```
PandoraApi::CaloHit::Create(m_pandora, caloHitParameters) throw STATUS_CODE_NOT_FOUND
    in function: CreateMuonCaloHits
    in file:     /tmp/root/spack-stage/spack-stage-ddmarlinpandora-0.12.01-p742e5lzpihslq2zfjsto66iki2aaskn/spack-src/src/DDCaloHitCreator.cc line#: 388
```
but not the `Warning: null charge particle`, nor the rest of the above `EDM4hepOutput` or `EventLoopMgr` errors/warnings. The application finally exits gracefully.

Comparing the root files (`events` tree), the bash version contains 3 events while the zsh version contains 2 events, and the branches also differ.

I'm a bit perplexed that the differences are only due to which shell is used... (I put the snippet below in a script, and run it with `./script.sh` or `bash script.sh`, from a fresh terminal)

Antoine

---

Steps to reproduce: follow the [ILD reco tutorial](https://key4hep.github.io/key4hep-doc/tutorials/key4hep-tutorials/gaudi_ild_reco/README.html) (Setup + Simu + Reco/Using `ILDReconstruction.py`).

- OS version: Alma9 (DESY NAF)
- key4hep stack: release 2024-04-12
- compiler, ROOT ...: shipped with key4hep stack.

```bash
source /cvmfs/sw.hsf.org/key4hep/setup.sh -r 2024-04-12

mkdir key4hep_tut_ild_reco
cd key4hep_tut_ild_reco

git clone https://github.com/iLCSoft/ILDConfig
cd ILDConfig/StandardConfig/production
wget https://raw.githubusercontent.com/key4hep/key4hep-tutorials/main/gaudi_ild_reco/input_files/zh_mumu.slcio

ddsim --compactFile $k4geo_DIR/ILD/compact/ILD_l5_v02/ILD_l5_v02.xml \
      --steeringFile ddsim_steer.py \
      --inputFiles zh_mumu.slcio \
      --outputFile zh_mumu_SIM.edm4hep.root

k4run ILDReconstruction.py \
      --detectorModel=ILD_l5_o1_v02 \
      --inputFiles=zh_mumu_SIM.edm4hep.root \
      --outputFileBase=zh_mumu
```

**Comment by alaudrain** [comment]
Hello again,

I'm closing this issue as results are not reproducible: running the same command multiple times in a fresh shell might or not yield the error.

Nevertheless: some precisions about the tests I conducted above.
- I figured out what I thought were clean shells might not have been: some had the key4hep release already setup before running the script. This should not matter as the release is setup in the script anyways, but still.
- As I didn't put a shabang at the beginning of the script, I assumed calling `./myscript.sh` would use the current shell (zsh in my case) as interpreter. After checking, in the absence of shabang, `sh` is rather called by default, which on my system is softlinked to `bash`. This makes even less sense as calling `./myscript.sh` and `bash myscript.sh` should then run the same interpreter (and hopefully give the same results)...
- To add even more to my confusion, I redid systematic checks using fresh shells and explicitly calling the script with `zsh`, `bash` or `sh`, which yield inconsistent results.
- I suspected environment variables or my zsh rc would be an issue so I tried the previous line adding `env -i` in front, and combining with passing the options `zsh -f` (to ignore rc files) and `bash --norc --noprofile`. Again the results are inconsistent.

TL;DR: I have no idea what cause this difference in the running of ILDReco.
